### ISSUE TITLE
Internal `$ref` resolution in a referenced external JSON Schema is incorrectly resolved against OAS context

### ISSUE THREAD

**Issue by mimkorn** [stoplightio/prism#2385]
I've encountered an issue with how Prism handles `$ref` resolution, especially when referencing external JSON Schema files that contain their own internal references.

## Context
When my OpenAPI specification references an external JSON Schema, and that external schema has its own internal references using the hash format (e.g., `"$ref": "#/definitions/PropertyType"`), it seems that Prism is attempting to resolve these internal references within the context of the main OpenAPI document rather than the external JSON Schema.

## Example

**OpenAPI Spec:**
```yaml
components:
  schemas:
    GenericContent:
      $ref: 'https://some.host/externally-hosted-schema.json#/definitions/Content'
```

**`externally-hosted-schema.json`:**
```json
{
  "$id": ".."
  "schema" : ".."
  "definitions": {
    "Content": {
      "additionalProperties":{
         "$ref": "#/definitions/PropertyType"
      }
      "type": "object",
      "properties": {
        ....
      }
    },
    "PropertyType": {
      "type": "string"
    }
  }
}
```

When Prism processes the above, it appears to look for `PropertyType` within the OpenAPI document instead of in the `external-schema.json` file where it is defined.

## Expected Behavior
Prism should resolve the internal `$ref` (pointing to `PropertyType`) relative to `external-schema.json`, not the main OpenAPI spec.

## Current Behavior
Prism is looking for `PropertyType` in the main OpenAPI spec instead of `external-schema.json` and fails to locate it:

```
MissingPointerError: at "#/components/schemas/GenericContent/additionalProperties", token "definitions" in "#/definitions/PropertyType" does not exist
```

I understand that the OpenAPI specification states reference resolution should follow the JSON Reference specification and not the JSON Schema specification. The behavior I'm seeing in Prism appears to be inconsistent with this.

Would appreciate any guidance or fixes related to this issue. Thank you for your attention and assistance.

## Possible Workaround/Solution
Currently a way to work around it seems to give absolute references instead of relative in the external files, but we don't like that much. It means providing the whole absolute web URL as a prefix for each internal reference.

## Environment
OpenAPI 3.0, 

- Version used: 5.3.1
- node v20.0.0
- Operating System and version Mac M1 Pro Ventura 13.5.2

**Comment by brendarearden** [comment]
Are you able to resolve your issue by updating your reference to the full json path in the external JSON Schema? 

Based on the example provided, `$ref: 'https://some.host/externally-hosted-schema.json#/Content'` should be `$ref: 'https://some.host/externally-hosted-schema.json#/definitions/Content'`. I see the path in the error output is different than the schema you have provided so this might not apply. 

I was unable to reproduce the `MissingPointerError` using the following schemas that are based on the example you provided: 

```
{
  "$id": "https://example.com/person.schema.json",
  "$schema": "https://json-schema.org/draft/2020-12/schema",
  "title": "Person",
  "type": "object",
  "definitions": {
    "Name": {
      "additionalProperties": {
        "$ref": "#/definitions/PropertyType"
      },
      "type": "object",
      "properties": {
        "id": {
          "type": "string"
        }
      }
    },
    "PropertyType": {
      "type": "string"
    }
  }
}

```

```
openapi: 3.0.0
info:
  title: example external schema
  version: '1.0'
servers:
  - url: 'http://localhost:3000'
paths:
  '/users/{userId}':
    parameters:
      - schema:
          type: integer
        name: userId
        in: path
        required: true
        description: Id of an existing user.
    get:
      summary: Get User Info by User ID
      tags: []
      responses:
        '200':
          description: User Found
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/User'
components:
  schemas:
    User:
      $ref: 'https://raw.githubusercontent.com/brendarearden/garbage/main/schemas/externally-hosted-schema-defs-only.json#/definitions/Name'

```

If changing the JSON path does not fix the issue for you, could you please provide a complete spec/schema that reproduces the issue?

**Comment by mimkorn** [comment]
Hi, yeah, I messed up, when I was providing the example. We do have it correctly referenced in the original file. So that is like you provided it: `$ref: 'https://some.host/externally-hosted-schema.json#/definitions/Content'` I edited my post. 

After some narrowing down it turns out it is the `x-examples` that messes things up. See below your modified example:

```yaml
openapi: 3.0.0
info:
  title: example external schema
  version: '1.0'
servers:
  - url: 'http://localhost:3000'
paths:
  '/users/{userId}':
    parameters:
      - schema:
          type: integer
        name: userId
        in: path
        required: true
        description: Id of an existing user.
    get:
      summary: Get User Info by User ID
      tags: []
      responses:
        '200':
          description: User Found
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/User'
components:
  schemas:
    User:
      $ref: 'https://raw.githubusercontent.com/brendarearden/garbage/main/schemas/externally-hosted-schema-defs-only.json#/definitions/Name'
      x-examples:
        Example 1: 
          property1 : "something"
```

I just added x-examples to your example and that's when you get the error:

`  stack: 'MissingPointerError: at "#/components/schemas/User/additionalProperties", token "definitions" in "#/definitions/PropertyType" does not exist\n' +`

**Comment by brendarearden** [comment]
@mimkorn the fix for this is in the latest version (v5.4.0) of prism and is also in the master branch!
